rust-router/src/route_filter.rs

```
@@ -17,6 +17,123 @@ fn project_to_meters(lon: f64, lat: f64, cos_ref_lat: f64) -> (f64, f64) {
+    let abx = bx - ax;
+    let aby = by - ay;
+    let ab2 = abx * abx + aby * aby;
+    if ab2 == 0.0 {
+        let dx = px - ax;
+        let dy = py - ay;
+        dx * dx + dy * dy
+    } else {
+        let apx = px - ax;
+        let apy = py - ay;
+        let mut t = (apx * abx + apy * aby) / ab2;
+        if t < 0.0 {
```

Clippy (1.95.0)

[warning] clamp-like pattern without using clamp function

(warning)